### PR TITLE
Add other autograd options for `AutoDiffCostFunction`

### PR DESCRIPTION
**Pull request by luisenp**

### CODE REVIEW

**Review by suddhu — APPROVED** [review]
Looks good to me, consistent with what I have!

File: theseus/core/cost_function.py [hunk 1]
```diff
@@ -207,8 +207,10 @@ def __init__(
         cost_weight: Optional[CostWeight] = None,
         aux_vars: Optional[Sequence[Variable]] = None,
         name: Optional[str] = None,
+        autograd_create_graph: bool = True,
         autograd_strict: bool = False,
         autograd_vectorize: bool = False,
+        autograd_strategy: str = "reverse-mode",
```
Comment (fantaosha) [inline theseus/core/cost_function.py:213]: Is `vmap` compatible with `forward-mode`?

File: theseus/core/cost_function.py [hunk 2]
```diff
@@ -207,8 +207,10 @@ def __init__(
         cost_weight: Optional[CostWeight] = None,
         aux_vars: Optional[Sequence[Variable]] = None,
         name: Optional[str] = None,
+        autograd_create_graph: bool = True,
         autograd_strict: bool = False,
         autograd_vectorize: bool = False,
```
Comment (fantaosha) [inline theseus/core/cost_function.py:212]: I think it is better to have `autograd_vectorize=True` by default.

**Review by fantaosha — APPROVED** [review]
Only minor comments. LGTM overall!

**Pull request merged**